File: solo.backend/Solo.Domain/Entities/ParkObject.cs
using System.Collections.Generic;
using Solo.Domain.Map;

namespace Solo.Domain.Entities
{
    public class ParkObject : EntityBase
    {
        public int ParkId { get; set; }
        public Park Park { get; set; }

        public string Name { get; set; }
        public string PublicDescriptionMarkdown { get; set; }
        public string AdministrationDescriptionMarkdown { get; set; }
        public Point Location { get; set; }
        public decimal PriceForAdults { get; set; }
        public decimal PriceForChildren { get; set; }
        public string ImageUrl { get; set; }
        public ObjectType Type { get; set; }

        /// <summary>
        /// <see cref="WorkSchedule"/>
        /// </summary>
        public string WorkScheduleJson { get; set; }

        public ICollection<Discount> Discounts { get; set; }

    }

    public enum ObjectType
    {
        None,
        Sight,
        Attraction,
        Shop,
        Food
    }
}

File: solo.frontend/src/api/ParkObject.ts
import { GeoPoint } from "./GeoPoint";

export interface ParkObject {
    id: number;
    parkId: number;
    name: string;
    publicDescriptionMarkdown: string;
    administrationDescriptionMarkdown: string;
    location: GeoPoint;
    priceForAdults: number;
    priceForChildren: number;
    workScheduleJson: string;
    imageUrl: string;
    type: ParkObjectType;
    ticketsTotal: number;
    ticketsClosed: number;
}

export enum ParkObjectType {
    None,
    Sight,
    Attraction,
    Shop,
    Food,
}
